src/osd/osd_types.h

```diff
@@ -1389,6 +1390,12 @@ struct pg_pool_t {
   /// application -> key/value metadata
   map<string, std::map<string, string>> application_metadata;
 
+  double mclock_res;
+  double mclock_wgt;
+  double mclock_lim;
+
+  crimson::dmclock::ClientInfo dmc_cli_info;
```

Comment:
It seems unnecessarily redundant to hold both the three doubles and the ClientInfo? What is the reason not to just keep the ClientInfo and get rid of the doubles?

Reply:
I removed the duplicate and changed it to use only ClientInfo.
The reason why the double and the ClientInfo were used in duplicate was that the role of op_class_client_info_f () changed from value to pointer return later.